internal/stack/serverless.go

```diff
@@ -0,0 +1,465 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package stack
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/elastic/elastic-package/internal/compose"
+	"github.com/elastic/elastic-package/internal/docker"
+	"github.com/elastic/elastic-package/internal/elasticsearch"
+	"github.com/elastic/elastic-package/internal/kibana"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/profile"
+	"github.com/elastic/elastic-package/internal/serverless"
+)
+
+const (
+	paramServerlessProjectID   = "serverless_project_id"
+	paramServerlessProjectType = "serverless_project_type"
+	paramServerlessFleetURL    = "serverless_fleet_url"
+
+	configRegion      = "stack.serverless.region"
+	configProjectType = "stack.serverless.type"
+
+	defaultRegion      = "aws-us-east-1"
+	defaultProjectType = "observability"
+)
+
+var (
+	allowedProjectTypes = map[string]struct{}{
+		"security":      {},
+		"observability": {},
+	}
```

Comment:
Nit. I would prefer this to be an array. Later you can use `common.StringSliceContains()` to check if a given one is one of the allowed ones.
```suggestion
	allowedProjectTypes = []string{
		"security",
		"observability",
	}
```